packages/pigeon/platform_tests/test_plugin/ios/Classes/CoreTests.gen.swift

```diff
@@ -3059,10 +3276,149 @@ class HostIntegrationCoreApiSetup {
     } else {
       echoAnotherAsyncNullableEnumChannel.setMessageHandler(nil)
     }
+    /// Returns the passed object, to test serialization and deserialization.
+    let taskQueue = binaryMessenger.makeBackgroundTaskQueue?()
+    let echoAllTypesTaskQueueBackgroundChannel = FlutterBasicMessageChannel(
+      name:
+        "dev.flutter.pigeon.pigeon_integration_tests.HostIntegrationCoreApi.echoAllTypesTaskQueueBackground\(channelSuffix)",
+      binaryMessenger: binaryMessenger,
+      codec: codec,
+      taskQueue: taskQueue
+    )
+    if let api = api {
+      echoAllTypesTaskQueueBackgroundChannel.setMessageHandler { message, reply in
+        let args = message as! [Any?]
+        let everythingArg = args[0] as! AllTypes
+        do {
+          let result = try api.echoAllTypesTaskQueueBackground(everythingArg)
+          reply(wrapResult(result))
+        } catch {
+          reply(wrapError(error))
```

Comment:
IIUC `reply` should still be called on main thread.

Reply:
Would it be sufficient to add?

```swift
DispatchQueue.main.async {
  reply(wrapResult(result))
}
```

like it is [mentioned in docs](https://docs.flutter.dev/platform-integration/platform-channels#jumping-to-the-main-thread-in-ios)

Reply:
However, I wonder why it was not done for others generators in the past.

Outputs for methods marked with `TaskQueue`:

Objective-c:
![image](https://github.com/user-attachments/assets/0110f03c-0b67-4633-81ee-6a467e9b0322)

Java:
![image](https://github.com/user-attachments/assets/d4b8b0ff-9f71-4d4c-902d-297f2675dd71)

Kotlin:
![image](https://github.com/user-attachments/assets/cfab70ea-f696-4165-8490-a49853090dc8)

Reply:
> IIUC `reply` should still be called on main thread.

Please see the second bullet point [here](https://github.com/flutter/packages/pull/8341#discussion_r1919146820).

Reply:
> > IIUC `reply` should still be called on main thread.
> 
> Please see the second bullet point [here](https://github.com/flutter/packages/pull/8341#discussion_r1919146820).

Sorry i misread your message.

Reply:
looking at it again, I must have misunderstood the 3rd bullet point as the `reply` call. Sorry for the confusion.

Reply:
> I must have misunderstood the 3rd bullet point as the `reply` call

Does it have any implications for the discussion in [this PR](https://github.com/flutter/packages/pull/8341)?

Apologies if I’m missing something.

Reply:
Yes, in my [previously proposed code](https://github.com/flutter/packages/pull/8341#discussion_r1919328977) I proposed to use an inner `Task` to switch back to main before calling `reply`, because I mistakenly thought `reply` has to be called on `main`. 

Anyways I will put up a design doc which is easier for us to discuss.